src/sagemaker/workflow/step_collections.py

```diff
@@ -227,6 +246,18 @@ def __init__(
                         inference_instances[0] if inference_instances else None
                     )
                 ]
+            if inference_recommender_params_exist(
+                framework, framework_version, nearest_model_name, data_input_configuration
+            ):
+                for container_obj in self.container_def_list:
+                    container_obj.update(
+                        update_container_object(
+                            framework,
+                            framework_version,
+                            nearest_model_name,
+                            data_input_configuration,
+                        )
+                    )
```

Comment:
Thanks for extracting helper functions. 

Can we combine the two functions `inference_recommender_params_exist` and `update_container_object` into a single function with name like `update_container_with_inference_params` or any other.

Within this combined method, it 
- firstly checks whether if the params exist, if no then return
- If exist, then iterate container_def_list to update each object

Seems currently inference_recommender_params_exist and update_container_object always show up together so we can put them into one function.

Reply:
Thanks for pointing this out.
I have refactored the code as per ask.